packages/@aws-cdk/aws-kinesisfirehose-alpha/test/delivery-stream.test.ts

```diff
@@ -151,12 +152,12 @@ describe('delivery stream', () => {
     Template.fromStack(stack).resourceCountIs('AWS::IAM::Role', 2);
   });
 
-  test('not providing role but specifying encryptionKey creates two roles', () => {
+  test('not providing role but using customerManagedKey encryption with a key creates two roles', () => {
     const key = new kms.Key(stack, 'Key');
 
     new firehose.DeliveryStream(stack, 'Delivery Stream', {
       destinations: [mockS3Destination],
-      encryptionKey: key,
+      encryption: StreamEncryption.customerManagedKey(key),
```

Comment:
This pattern feels a bit off to me. `StreamEncryption` usage sounds like a ENUM to me but the actual usage is calling a statis method.

Reply:
This is the same pattern that's being used for TableV2's encryption [here](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-dynamodb/README.md#encryption).